File: src/main/java/space/accident/api/interfaces/tileentity/redstone/IRedStoneReceiver.java
package space.accident.api.interfaces.tileentity.redstone;

import space.accident.api.interfaces.tileentity.IWorldInteraction;

/**
 * This File has just internal Information about the RedStone State of a TileEntity
 */
public interface IRedStoneReceiver extends IWorldInteraction {
	/**
	 * gets the RedStone Level of the TileEntity to the given Input Side
	 * <p/>
	 * Do not use this if ICoverable is implemented. ICoverable has @getInternalInputRedstoneSignal for Machine internal Input RedStone
	 * This returns the true incoming RedStone Signal. Only Cover Behaviors should check it, not MetaTileEntities.
	 */
	int getInputRedStoneSignal(int side);
	
	/**
	 * gets the strongest RedStone Level the TileEntity receives
	 */
	int getStrongestRedStone();
	
	/**
	 * gets if the TileEntity receives RedStone
	 */
	boolean getRedStone();
	
	/**
	 * gets if the TileEntity receives RedStone at this Side
	 */
	boolean getRedStone(int side);
}

File: src/main/kotlin/space/accident/api/interfaces/te/IRedstoneReceiver.kt
package space.accident.api.interfaces.te

/**
 * This File has just internal Information about the Redstone State of a TileEntity
 */
interface IRedstoneReceiver : IHasWorldObjectAndCoords {
    /**
     * gets the Redstone Level of the TileEntity to the given Input Side
     *
     *
     * Do not use this if ICoverable is implemented. ICoverable has @getInternalInputRedstoneSignal for Machine internal Input Redstone
     * This returns the true incoming Redstone Signal. Only Cover Behaviors should check it, not MetaTileEntities.
     */
    fun getInputRedstoneSignal(aSide: Byte): Byte

    /**
     * gets the strongest Redstone Level the TileEntity receives
     */
    fun getStrongestRedstone(): Byte

    /**
     * gets if the TileEntity receives Redstone
     */
    fun getRedstone(): Boolean

    /**
     * gets if the TileEntity receives Redstone at this Side
     */
    fun getRedstone(aSide: Byte): Boolean
}